### ISSUE TITLE
False doc_markdown warnings

### ISSUE THREAD

**Issue by compressed** [Manishearth/rust-clippy#1469]
When I use `cfg_attr` for conditional examples in doc comments like this:

```rust
#![cfg_attr(feature = "a", doc = " ```")]
#![cfg_attr(not(feature = "a"), doc = " ```ignore")]
//! fn main() {
//!     let s = "localhost:10000".to_string();
//!     println!("{}", s);
//! }
//! ```
```

I receive false warnings:

```
cargo clippy --features a
...
warning: you should put `to_string` between ticks in the documentation, #[warn(doc_markdown)] on by default
  --> src/lib.rs:66:35
   |
66 | //!     let s = "localhost:10000".to_string();
   |                                   ^^^^^^^^^
```

**Comment by oli-obk** [comment]
I wasn't even aware that's possible. That's amazing...

**Comment by compressed** [comment]
Yeah, I had to learn about how to do it correctly in the first place 😄. Definitely an edge case.

**Comment by mcarton** [comment]
This is such a hack, I hesitate between :smile: and 😱.

**Comment by Manishearth** [comment]
I think this is a niche enough use case that clippy shouldn't attempt to handle it. idk. Ultimately, the way clippy is supposed to be used is with a liberal sprinkling of allows.

**Comment by compressed** [comment]
I'd be fine putting an `allow`, but since it's at the module/crate level, I'm not sure how to just have it apply there.

If I put `#![cfg_attr(feature = "cargo-clippy", allow(doc_markdown))]`, I'd turn off the validation for the entire crate, not just this section. Maybe there is a better workaround?

**Comment by iliekturtles** [comment]
I ran into the same issue at the [crate level](https://github.com/iliekturtles/uom/blob/09bfe4eafa04fef22e72a43b59dcfde70bd9ad7a/src/lib.rs#L33) and I'm hesitant to add `#![allow(doc_markdown)]`. I briefly attempted to do a github-wide search for other repositories to see how many others use this same functionality, but didn't get very far.
